staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -5241,6 +5241,8 @@ type ReplicationController struct {
 	// +optional
 	// +k8s:subfield(name)=+k8s:optional
 	// +k8s:subfield(name)=+k8s:format=dns-subdomain
+	// +k8s:subfield(generateName)=+k8s:optional
+	// +k8s:subfield(generateName)=+k8s:format=generate-name(dns-subdomain)
```

Comment:
WIP: pending decision on tag style

Reply:
The composition here is looking really good to me.

I just realized earlier that we can map `subfield to JSON Schema "allOf" when translating this to OpenAPI.

(I've commented elsewhere on tag style topic).